src/components/RefinementList/__tests__/RefinementList-test.js

```diff
@@ -10,176 +10,318 @@ import expectJSX from 'expect-jsx';
-      let customProps = {
+  describe('sublist', () => {
+    it('should create a subList with the sub values', () => {
+      // Given
+      let props = {
+        facetValues: [
+          {
+            name: 'foo',
+            data: [
+              {name: 'bar'},
+              {name: 'baz'}
+            ]
+          }
+        ]
+      };
+
+      // When
+      let root = shallowRender(props);
+      let mainItem = root.find(RefinementListItem).at(0);
+      let subList = shallow(mainItem.props().subItems);
+      let subItems = subList.find(RefinementListItem);
+
+      // Then
+      expect(mainItem.props().facetValueToRefine).toEqual('foo');
+      expect(subItems.at(0).props().facetValueToRefine).toEqual('bar');
+      expect(subItems.at(1).props().facetValueToRefine).toEqual('baz');
+    });
+
+    it('should add depth class for each depth', () => {
+      // Given
+      let props = {
         cssClasses: {
-          depth: 'depth',
-          item: 'item',
-          list: 'list',
-          active: 'active'
+          depth: 'depth-'
         },
         facetValues: [
           {
-            name: 'facet1',
-            isRefined: true,
+            name: 'foo',
             data: [
-              {name: 'subfacet1'},
-              {name: 'subfacet2'}
+              {name: 'bar'},
+              {name: 'baz'}
             ]
           }
         ]
       };
-      parentListProps = {
-        className: 'list depth0'
-      };
-      itemProps[0].templateData.cssClasses = customProps.cssClasses;
-      itemProps[0].templateData = {
-        ...itemProps[0].templateData,
-        data: customProps.facetValues[0].data
-      };
-      itemProps[0].subItems = (
-        <RefinementList
-          attributeNameKey="name"
-          createURL={() => {}}
-          cssClasses={customProps.cssClasses}
-          depth={1}
-          facetValues={customProps.facetValues[0].data}
-          templateProps={{}}
-        />
-      );
-      let out = render(customProps);
-      expect(out).toEqualJSX(
-        <div {...parentListProps}>
-          <RefinementListItem
-            {...itemProps[0]}
-          />
-        </div>
-      );
-    });
-  });
 
-  function render(extraProps = {}) {
-    let props = getProps(extraProps);
-    renderer.render(<RefinementList {...props} ref="list" templateProps={{}} />);
-    return renderer.getRenderOutput();
-  }
+      // When
+      let root = shallowRender(props);
+      let mainItem = root.find(RefinementListItem).at(0);
+      let subList = shallow(mainItem.props().subItems);
 
-  function getProps(extraProps = {}) {
-    return {
-      cssClasses: {
-        list: 'list',
-        item: 'item',
-        active: 'active'
-      },
-      facetValues: [
-        {name: 'facet1', isRefined: true},
-        {name: 'facet2', isRefined: false}
-      ],
-      createURL: () => 'url',
-      ...extraProps
-    };
-  }
+      // Then
+      expect(root.props().className).toContain('depth-0');
```

Comment:
I feel we are loosing a bit of testing surface here because we are only testing we have a subItems object that has a className equals to depth-1, while before that we were testing more than that?